src/components/AlertDialog.tsx

```diff
@@ -24,30 +31,19 @@ export default function AlertDialog(props: AlertProps) {
   };
 
   async function handleAgree() {
+    setLoading(true);
     try {
-      await deleteById(props.id);
-      toast.success("Event Deleted Successfully! ", {
-        position: "top-right",
-        autoClose: 5000,
-        hideProgressBar: false,
-        closeOnClick: true,
-        pauseOnHover: true,
-        draggable: true,
-        theme: "colored",
-      });
+      await deleteById(id);
+      const newEvents = events?.filter((item) => item.id != id);
+      dispatch(setEvents(newEvents));
+      setLoading(false);
```

Comment:
@LaithAbusada move this state handler to a `finally` block; if the `deleteById` request fails, the loader will get stuck, because `setLoading` will not run. Check this [MDN page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch).

Reply:
Yes i will change it right now, the setOpen(false) closes the dialog, so the loading state that is in the dialog would not matter , as it is now closed